lib/demos/material/dialog_demo.dart

```diff
@@ -35,101 +90,148 @@ class DialogDemo extends StatelessWidget {
     return '';
   }
 
-  Future<void> _showDemoDialog<T>({BuildContext context, Widget child}) async {
-    child = ApplyTextOptions(
-      child: Theme(
-        data: Theme.of(context),
-        child: child,
-      ),
-    );
-    final value = await showDialog<T>(
-      context: context,
-      builder: (context) => child,
-    );
-    // The value passed to Navigator.pop() or null.
-    if (value != null && value is String) {
-      ScaffoldMessenger.of(context).hideCurrentSnackBar();
-      ScaffoldMessenger.of(context).showSnackBar(SnackBar(
-        content:
-            Text(GalleryLocalizations.of(context).dialogSelectedOption(value)),
-      ));
-    }
-  }
-
-  void _showAlertDialog(BuildContext context) {
+  static Route<String> _alertDialogDemoRoute(
+    BuildContext context,
+    Object arguments,
+  ) {
     final theme = Theme.of(context);
     final dialogTextStyle = theme.textTheme.subtitle1
         .copyWith(color: theme.textTheme.caption.color);
-    _showDemoDialog<String>(
+
+    final themes = InheritedTheme.capture(
+      from: context,
+      to: Navigator.of(
+        context,
+        rootNavigator: true,
+      ).context,
+    );
+
+    return DialogRoute<String>(
       context: context,
-      child: AlertDialog(
-        content: Text(
-          GalleryLocalizations.of(context).dialogDiscardTitle,
-          style: dialogTextStyle,
+      themes: themes,
+      builder: (context) => ApplyTextOptions(
+        child: Theme(
+          data: Theme.of(context),
```

Comment:
Why this? Isn't this just wrapping everything in the same ThemeData that children could access via `Theme.of` even if it wasn't re-wrapped like this?

Reply:
Same question for the instances below.